pkg/controller/apicuriodeployment/apicuriodeployment_controller.go

```diff
@@ -0,0 +1,257 @@
+package apicuriodeployment
+
+import (
+	"context"
+	"log"
+
+	integreatlyv1alpha1 "github.com/integr8ly/apicurio-operator/pkg/apis/integreatly/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"fmt"
+	openshift "github.com/integr8ly/apicurio-operator/pkg/apis/integreatly/openshift/client"
+	v1template "github.com/openshift/api/template/v1"
+	kuberr "k8s.io/apimachinery/pkg/api/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+	"github.com/openshift/api/image/v1"
+	v12 "github.com/openshift/api/apps/v1"
+)
+
+// Add creates a new ApicurioDeployment Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileApiCurioDeployment{
+		client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+		config: mgr.GetConfig(),
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	opts := controller.Options{Reconciler: r}
+	c, err := controller.New("apicuriodeployment-controller", mgr, opts)
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource ApicurioDeployment
+	err = c.Watch(&source.Kind{Type: &integreatlyv1alpha1.ApicurioDeployment{}}, &handler.EnqueueRequestForObject{})
```

Comment:
checks like these can be simplified 
``` 
if err := function(); err != nil{
    return err
}

Reply:
That's generated code https://github.com/operator-framework/operator-sdk/blob/4bf0c692308ee6591374c602c6635014cbc06158/pkg/scaffold/controller_kind.go#L91